doc/terminology.md

```diff
@@ -0,0 +1,86 @@
+
+A client is a program that intends to consume the VSS data points. This includes reading of sensors or actuating actors. A client intends to use capabilites offered by a vehicle.
+
+So typically this might be "apps" running on vehicle computers, the infotainment, or maybe even a user's personal device. 
+
+### Providers
+A provider is a program that makes sure interactions with the VSS data points are linked to the state of a vehicle.
+
+A provider can can supply either be a
+ *  *sensor-provider*, providing information for a sensor defined in VSS, i.e. a component that updates `Vehicle.Speed` based on the actual speed of an vehicle.
+ * An *actuator-provider*, trying to act on the target value of an actuator defined in VSS, i.e. a component that based on the target value of `Vehicle.Body.Trunk.Rear.IsOpen` tries to ensure that the actual state of the trunk (open or closed) matches the desired state set by clients.
+
+ ## Vehicle Signal Specification (VSS)
+ KUKSA.val handles datapoints that are defined using the COVESA Vehicle Specification (VSS).
+
+ VSS introduces a domain taxonomy for vehicle signals. It defines
+ * A syntax for defining vehicle signals in a structured manner.
+ * A catalog of standard signals (VSs standard catalogue) related to vehicles.
+
+It can be used as standard in automotive applications to communicate information around the vehicle, which is semantically well defined. 
+
+See the [the official VSS documentation](https://covesa.github.io/vehicle_signal_specification/) for all details. Here we will summarizes the terms and concepts most relevant for KUKSA.val.
+
+### Signal
+Everything we handle in KUKSA.val is defined as a using VSS. A signal in VSS is an entity identified by a dot-separated string in VSS, i.e. `Vehicle.Speed`.
+
+A signal can be of different types. VSS defines the following signal types:
+ - *Sensor*: Sensors are signals to read values of properties in a vehicle. Values of sensors typically change over time. Reading a sensor shall return the current actual value of the related property, e.g. the current speed of a vehicle. Example: `Vehicle.Speed`. [[Source]](https://covesa.github.io/vehicle_signal_specification/rule_set/data_entry/sensor_actuator/)
+ - *Actuator*: Actuators are signals that are  used to control the desired value of a property. _Every Actuator in VSS is also a Sensor_. Some properties in a vehicle may change instantly. A typical example is position of a seat. Reading a value of an actuator shall return the current actual value (i.e. the sensor trait of that actuator signal), e.g. the current position of the seat, rather than the wanted/desired position. Example: `set Vehicle.Cabin.Seat.Row1.Pos1.Position`. [[Source]](https://covesa.github.io/vehicle_signal_specification/rule_set/data_entry/sensor_actuator/)
+ KUKSA.val fully supports actuators, and alwaysgives you full control whether you are interacting with the _target value_ or the _current value_ of an actuator signal. 
```

Comment:
Not sure what this sentence adds in terms of terminology, really.

Reply:
Guess this refers to the "advertising language" -> removed